bootloader/src/pyi_main.c

```diff
@@ -163,6 +185,19 @@ pyi_main(int argc, char * argv[])
 
         VS("LOADER: set _MEIPASS2 to %s\n", pyi_getenv("_MEIPASS2"));
 
+#if defined(__linux__)
+
+        /* Pass the process name to child via environment variable. */
+        if (1) {
+            char processname[16]; /* 16 bytes as per prctl() man page */
+            if (!prctl(PR_GET_NAME, processname, 0, 0)) {
```

Comment:
Where did you find that? The man page on my system says only this:
```
PR_GET_NAME (since Linux 2.6.11)
              Return  the name of the calling thread, in the buffer pointed to
              by (char *) arg2.  The buffer should allow space for  up  to  16
              bytes; the returned string will be null-terminated.
```
and the entry for SET_NAME seems to suggest that null byte is included in those 16 bytes.

This is also in line with pyinstaller's own test on process name (`test_symlink_basename_is_kept`), which check that 15-character process name is preserved. So it should be 15+1.